packages/strapi-plugin-content-type-builder/services/ContentTypeBuilder.js

```diff
@@ -44,18 +44,23 @@ module.exports = {
 
     const model = source ? _.get(strapi.plugins, [source, 'models', name]) : _.get(strapi.models, name);
 
-    // const model = _.get(strapi.models, name);
-
     const attributes = [];
     _.forEach(model.attributes, (params, name) => {
       const relation = _.find(model.associations, { alias: name });
 
-      if (relation) {
-        params = _.omit(params, ['collection', 'model', 'via']);
-        params.target = relation.model || relation.collection;
-        params.key = relation.via;
-        params.nature = relation.nature;
-        params.targetColumnName = _.get((params.plugin ? strapi.plugins[params.plugin].models : strapi.models )[params.target].attributes[params.key], 'columnName', '');
+      if (relation &&  !_.isArray(_.get(relation, 'related'))) {
```

Comment:
This part seems hardcoded?

```js
_.get(relation, 'related')
```

The user can edit the relation attribute name.